sunpy/map/sources/punch.py

```diff
@@ -0,0 +1,51 @@
+"""PUNCH Map subclass definitions"""
+
+from matplotlib.colors import LogNorm
+
+import astropy.units as u
+
+from sunpy.map.mapbase import GenericMap
+
+__all__ = ['PUNCHMap']
+
+class PUNCHMap(GenericMap):
+    """
+    PUNCH Map.
+
+    The Polarimeter to Unify the Corona and Heliosphere (PUNCH) is a constellation observatory consisting of four satellites, together imaging the solar corona in polarized white light.
+    Coordinated observations from three Wide Field Imagers (WFIs) and one Near Field Imager (NFI) are processed and meshed into one virtual observatory with a field of view spanning 45-degrees in radius away from the Sun.
+
+    PUNCH launched and began operations on 11 March 2025.
+
+    Notes
+    -----
+
+    References
+    ----------
+    * `PUNCH Mission Page <https://punch.space.swri.edu>`__
+    * `PUNCH SOC Development <https://github.com/punch-mission/>`__
+    """
+
+    def __init__(self, data, header, **kwargs):
+        super().__init__(data, header, **kwargs)
+        self.nickname = f"{self.observatory} - {self.instrument}"
+        self.plot_settings["cmap"] = "punch"
+
+        self.plot_settings["norm"] = LogNorm()
```

Comment:
```suggestion
        self.plot_settings["norm"] = ImageNormalize(
            stretch=source_stretch(self.meta, LogStretch()), clip=False)
```
Sorry for the additional suggestion. Better to use normalizations provided by astropy. You'll also need:

```python
from astropy.visualization import ImageNormalize, LogStretch
```

Reply:
Oh, no worries at all - we really welcome the best practice suggestions here!